File: tiq-e/array/3/main.ts
function rotate(nums: number[], k: number): void {
  nums.unshift(...nums.splice(nums.length - k % nums.length));
}

let x = [1, 2, 3, 4, 5, 6, 7]
rotate(x, 3)
console.log(x)

const iterations = 1000000;
console.time('Function #1');
for (let i = 0; i < iterations; i++) {
  rotate(x, 3);
}
console.timeEnd('Function #1')

File: tiq-e/array/3/main.go
package main

import (
	"fmt"
	"time"
)

func rotate(nums []int, k int) {
	l := len(nums)
	k = k % l
	newArr := append(nums[l-k:l], nums[0:l-k]...)
	copy(nums, newArr)
}

func main() {
	arr := []int{1, 1, 2, 3, 4}

	var start = time.Now()
	rotate(arr, 3)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", arr)
}
